src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -186,6 +204,23 @@ private function createRelationsForItem($item, $formattedData)
                     $modelInstance = new $model($relationData['values']);
                     $relation->save($modelInstance);
                 }
+            } elseif ($relation instanceof HasMany || $relation instanceof MorphMany) {
+                $keys = Arr::pluck($item->{$relationMethod}, $relation->getRelated()->getKeyName());
+                foreach (json_decode($relationData['values'][$relationMethod], true) as $relationItems) {
+                    if ($relationItems[$relation->getForeignKeyName()] != '' && $relationItems[$relation->getRelated()->getKeyName()] != '') {
+                        $modelInstance = $model::find($relationItems[$relation->getRelated()->getKeyName()]);
+                        $modelInstance->update($relationItems);
+                        if (($key = array_search($relationItems[$relation->getRelated()->getKeyName()], $keys)) !== false) {
+                            unset($keys[$key]);
+                        }
+                    } else {
+                        $modelInstance = new $model($relationItems);
+                        $relation->save($modelInstance);
+                    }
+                }
+                foreach ($keys as $id) {
+                    $item->{$relationMethod}->find($id)->delete();
```

Comment:
I wonder if it wouldn't simplify the code here to use Eloquent's `sync()` instead of `save()` and `delete()`. At first look it should reduce the number of conditionals and loops.